File: Patio-El-Olvidado/src/types/pago.ts
export type PagoMetodo = 'Efectivo' | 'Tarjeta' | 'Transferencia'
export type PagoEstado = 'Completado' | 'Anulado'

export interface Pago {
  id: number
  pedidoId: number
  metodo: PagoMetodo | string
  estado: PagoEstado | string
  monto: number
  fechaPago: string
  cajaId: number
}

export interface CreatePagoPayload {
  pedidoId: number
  metodo: PagoMetodo
  monto: number
}

export interface CajaDia {
  fecha: string
  totalEfectivo: number
  totalTarjeta: number
  totalTransferencia: number
  total: number
}

File: API-Patio-El-Olvidado/src/PatioElOlvidado.Domain/Entities/Pago.cs
namespace PatioElOlvidado.Domain.Entities;

/// <summary>Cobro parcial o total de un pedido (RF-04 / CU04).</summary>
public class Pago
{
    public int Id { get; set; }
    public int PedidoId { get; set; }
    /// <summary>Efectivo | Tarjeta | Transferencia</summary>
    public string Metodo { get; set; } = string.Empty;
    /// <summary>Completado | Anulado</summary>
    public string Estado { get; set; } = string.Empty;
    public decimal Monto { get; set; }
    public DateTime FechaPago { get; set; }
    public int CajaId { get; set; }

    public Pedido Pedido { get; set; } = null!;
    public Caja Caja { get; set; } = null!;
}
